sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/EpochCoordinator.scala

```diff
@@ -267,4 +275,16 @@ private[continuous] class EpochCoordinator(
       queryWritesStopped = true
       context.reply(())
   }
+
+  private def checkTotalCores(): Unit = {
+    val numExecutors = session.conf.get("spark.executor.instances", "1").toInt
+    val coresPerExecutor = session.conf.get("spark.executor.cores", "1").toInt
+    val totalCores = numExecutors * coresPerExecutor
+    logDebug(s"Check total cores $totalCores and kafka partition number $numReaderPartitions")
```

Comment:
Don't we need to consider `spark.task.cpus`? I think the actual number of tasks is decided by executor instance, cores and task cpus, right?

Reply:
Updated.